### DISCUSSION TITLE
Access other fields of the item in your custom interface

### DISCUSSION THREAD

**Discussion by nickluger** [directus/directus#8232] [Archive]
### Within a **custom interface**: 

Is there a way to gain access to other fields of an item within your custom interface, besides the specific field being edited?

This would be, for example, useful, if a custom interface should render or do something depending on the **dynamic value set in another field**.

Of course, fields already have access to other fields within the conditional-fields-rules, i.e. can be hidden or disabled based on other fields, but custom interfaces have no access to other fields (as far as I can see) of the containing item, though they may depend on it.

In our case, we e.g. fetch some data depending on the ID set in another field. Something like this is not handled by conditional rendering.

I took a look at `this.$attrs` and it seems no model data like `modelValue` or `initialValues` is accessible by custom interfaces. If I overlooked something, pls just show me where I can access these. 

### Workaround
There's a super-ugly workaround to get access to this:

```
this?.$parent?.$parent?.$parent?.$props?.modelValue?
//or
this?.$parent?.$parent?.$parent?.$props?.initialValues?
```

The problem being (besides the obvious ugliness) is that these are not props and your custom interface is not informed when they change. You can hack that by adding a conditional render, that does basically nothing, but re-renders the component and something like:

```
 watch: {
    fieldData() {
      // use $parent.$parent.. .to check whether you need to act or not
    },
  },
```

**Comment by rijkvanzanten** [comment]
The v-form currently uses "provide" to provide the full "values" object of the current form context to all interfaces. You should be able to use `const values = inject('values');` to get access to the reactive values object 👍🏻

**Reply by nickluger** [reply]
Perfect, that worked! Thank you!
